packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -343,17 +382,30 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
     xcodeProjectSettingsMap['Deployment Target'] = globals.plistParser.getStringValueFromFile(plistPath, PlistParser.kMinimumOSVersionKey);
     xcodeProjectSettingsMap['Bundle Identifier'] = globals.plistParser.getStringValueFromFile(plistPath, PlistParser.kCFBundleIdentifierKey);
 
-    xcodeProjectSettingsMap.forEach((String title, String? info) {
-      messageBuffer.writeln('$title: ${info ?? "Missing"}');
-    });
-
-    if (xcodeProjectSettingsMap.values.any((String? element) => element == null)) {
-      messageBuffer.writeln('\nYou must set up the missing settings.');
-    }
-
-    if (xcodeProjectSettingsMap['Bundle Identifier']?.startsWith('com.example') ?? false) {
-      messageBuffer.writeln('\nWarning: Your application still contains the default "com.example" bundle identifier.');
-    }
+    final List<ValidationMessage?> validationMessages = xcodeProjectSettingsMap.entries.map((MapEntry<String, String?> entry) {
+      final String title = entry.key;
+      final String? info = entry.value;
+      // message for both valid and invalid case.
+      final String message = '$title: ${info ?? "Missing"}';
+      return _createValidationMessage(
+        isValid: info != null,
+        messageIfValid: message,
+        messageIfInvalid: message);
+    }).toList();
+
+    final bool hasMissingSettings = xcodeProjectSettingsMap.values.any((String? element) => element == null);
+    validationMessages.add(_createValidationMessage(
+      isValid: !hasMissingSettings,
+      messageIfValid: null, // do not print out for the valid case.
+      messageIfInvalid: 'You must set up the missing app settings.'));
+
+    final bool usesDefaultBundleIdentifier = xcodeProjectSettingsMap['Bundle Identifier']?.startsWith('com.example') ?? false;
+    validationMessages.add(_createValidationMessage(
+      isValid: !usesDefaultBundleIdentifier,
+      messageIfValid: 'Your application does not contain the default "com.example" bundle identifier.',
```

Comment:
This also has too many negatives, it sounds like you should have the default but don't.  Again, suggest leaving off the success case and just printing the errors.

Reply:
Gotcha. In this case i should probably rename the section title "Image Assets Validation" to "App icon and launch image assets validation", so that when everything is valid, it's more clear what validation was performed.

Reply:
I think i should probably skip `Launch image is not set to the default placeholder image.` too, since it also sounds like an error.